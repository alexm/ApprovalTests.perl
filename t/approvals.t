#! perl
use strict;
use warnings FATAL => qw(all);
use autodie;
use version; our $VERSION = qv('v0.0.4_6');

use Test::Approvals::Specs qw(describe it run_tests);
use Test::Approvals qw(verify use_reporter reporter use_name use_name_in namer);
use Test::More;

describe 'An Approval', sub {
    use_reporter('Test::Approvals::Reporters::FakeReporter');
    it 'Reports failure using the configured reporter', sub {
        my ($spec) = @_;

        use_name($spec);
        verify('Hello');
        ok reporter()->was_called, $spec;
        unlink namer()->get_received_file('txt');
    };

    it 'Can generate names in any directory' => sub {
        my ($spec) = @_;
        my $expected =
            'C:\tmp\approvals.t.'
          . 'an_approval_can_generate_names_in_any_directory'
          . '.received.txt';
        use_name_in( $spec, 'C:\tmp' );
        is namer()->get_received_file('txt'), $expected, namer()->name;
      };

      it 'Still provides expected name for names in other directory' => sub{
      	my ($spec)=@_;
      	use_name_in($spec, 'C:\tmp');
      	is $spec, namer()->name, $spec;
      };
};

run_tests();
