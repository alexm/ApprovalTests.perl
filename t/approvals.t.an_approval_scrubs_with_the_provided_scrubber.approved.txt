$VAR1 = {
          'fname' => 'red',
          'lname' => 'lintrock',
          'ssn' => 'ABC123'
        };
